python/tchannel/messages/call_response.py

```diff
@@ -27,3 +32,87 @@ class CallResponseMessage(CallRequestMessage):
         'arg_2',
         'arg_3',
     )
+
+    FLAGS_SIZE = 1
+    CODE_SIZE = 1
+    # span_id + parent_id + trace_id = 3*8 = 24
+    TRACE_SIZE = 8
+    TRACEFLAGS_SIZE = 1
+    NH_SIZE = 1
+    HEADER_SIZE = 1
+    CSUMTYPE_SIZE = 1
+    CSUM_SIZE = 4
+    ARG_SIZE = 2
+
+    def parse(self, payload, size):
+        """Parse a call request message from a payload."""
+        self.flags = read_number(payload, self.FLAGS_SIZE)
+        self.code = read_number(payload, self.CODE_SIZE)
+
+        self.span_id = read_number(payload, self.TRACE_SIZE)
+        self.parent_id = read_number(payload, self.TRACE_SIZE)
+        self.trace_id = read_number(payload, self.TRACE_SIZE)
+
+        self.traceflags = read_number(payload, self.TRACEFLAGS_SIZE)
+
+        self.headers, _ = self._read_headers(
+            payload,
+            self.NH_SIZE,
+            self.HEADER_SIZE,
+        )
+
+        self.checksum_type = read_number(payload, self.CSUMTYPE_SIZE)
+        if self.checksum_type:
+            self.checksum = read_number(payload, self.CSUM_SIZE)
```

Comment:
Checksum size is not exactly to 4 bytes. It just so happens that all supported checksum types are 32-bit. use a mapping from checksum_type to checksum size (might as well update call req while you're at it).